core/analyze.go

```diff
@@ -0,0 +1,145 @@
+	url := fmt.Sprintf("https://%s/analysis/account", apiHost)
+
+	requestBody := analyzeRequestBody{
+		CustomerID: customerID,
+		Account:    account,
+	}
+	requestBodyBytes, _ := json.Marshal(requestBody)
+
+	request, err := http.NewRequest("POST", url, bytes.NewBuffer(requestBodyBytes))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not build API request: %s\n", err)
+		return err
+	}
+	now := time.Now()
+	request.Header.Set("Date", now.Format(time.RFC3339))
+	request.Header.Set("Content-Type", "application/json")
+
+	if err != signApiRequest(cfg, request, now) {
+		fmt.Fprintf(os.Stderr, "Could not sign API request: %s\n", err)
+		return err
+	}
+
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not execute API request: %s\n", err)
+		return err
+	} else {
+		defer response.Body.Close()
+	}
+
+	body, _ := ioutil.ReadAll(response.Body)
+	responseBodyStr := string(body)
+	if response.StatusCode == http.StatusOK || response.StatusCode == http.StatusAccepted {
+		analyzeResponse := analyzeResponseBody{}
+		if err != json.Unmarshal(body, &analyzeResponse) {
+			fmt.Fprintf(os.Stderr, "Could not deserialize API response: %s\n", responseBodyStr)
+			return err
+		}
+
+		fmt.Fprintf(o, "Started analysis for %s account %s with execution ID: %s\n",
+			customerID,
+			account,
+			analyzeResponse.ExecutionID)
+	} else {
+		fmt.Fprintf(os.Stderr,"Analyze API Response Status: %s\n", response.Status)
+		fmt.Fprintf(os.Stderr, "Could not start analysis for %s account %s.  API Response: %s\n",
+			customerID,
+			account,
+			responseBodyStr)
+	}
+	return nil
+}
+
+// Sign a request to the k9 Security AWS API gateway endpoint with an AWS v4 signature
+// using the aws-sdk-go-v2 v4.SignHTTP function.
+//
+// The request should be fully-built and ready to send to the gateway.
+//
+// The provided request will be modified in place.
+func signApiRequest(cfg aws.Config, request *http.Request, signingTime time.Time) error {
+	ctx := context.TODO()
+	credentials, err := cfg.Credentials.Retrieve(ctx)
+	if err != nil {
+		return err
+	}
+
+	hash, err := createPayloadHash(request)
+	if err != nil {
+		return err
+	}
+
+	signer := v4.NewSigner()
+	if err != signer.SignHTTP(ctx, credentials, request, hash, "execute-api", cfg.Region, signingTime) {
+		return err
+	}
+
+	// fmt.Printf("Authorization: %s\n", request.Header.Get("Authorization"))
+	return err
+}
+
+// Create a hex-encoded SHA256 hash of the request payload.  Use when calculating an AWS v4 signature.
+func createPayloadHash(req *http.Request) (string, error) {
+	// from https://github.com/yuizho/salon/blob/a159bcfeb263cb6502403f682c138286a2d7bb1f/backend/lambda/mutate-user/appsync/client.go
+	body, err := req.GetBody()
+	if err != nil {
+		return "", err
+	}
+	var buf bytes.Buffer
+	_, err = buf.ReadFrom(body)
+	if err != nil {
+		return "", err
+	}
```

Comment:
You could collapse this as well, but again, it is not super necessary.  Missed it on my first pass.